src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -247,11 +253,11 @@ const DiscretisedDensityOnCartesianGrid<3,float>* current_anatomical_cast =
 
     if(num_non_zero_feat>1){
          for(int i = 0; i <=anatomical_image_filenames.size()-1; i++){
-           this->kmnorm_sptr[i] = shared_ptr<TargetT>(this->anatomical_prior_sptr[i]->get_empty_copy ());
+           this->kmnorm_sptr.push_back( shared_ptr<TargetT>(this->anatomical_prior_sptr[i]->get_empty_copy ()));
```

Comment:
I haven't check this, but the previous version assigned to a specific index, while the new version will just grow. If this new behaviour is correct, it implies that the previous one should have crashed rather dramatically...

Reply:
Yes in fact, the previous version will fail if you use non_zero_element>1. sadly I noticed while running now

Reply:
ok. it's in a separate commit, so that's fine. However, now you have to be sure that `kmnorm_sptr` is empty when you start (maybe that's guaranteed, I don't know)

Reply:
mh not sure here. That line is the very first call to kmnorm_sptr. I assumed that would be empty until that point.